### PR TITLE
Remove all obsolete members from Input namespace

### PR DESCRIPTION
**Pull request by maxkatz6**
## Breaking changes

Previously obsolete members were removed.

## Fixed issues

Part of #3538

### CODE REVIEW

File: src/Avalonia.Controls/TopLevel.cs [hunk 1]
```diff
@@ -302,9 +304,6 @@ public ILayoutManager LayoutManager
             set { SetValue(PointerOverElementProperty, value); }
         }
 
-        /// <inheritdoc/>
-        IMouseDevice? IInputRoot.MouseDevice => PlatformImpl?.MouseDevice;
```
Comment (maxkatz6) [inline src/Avalonia.Controls/TopLevel.cs:306]: MouseDevice wasn't obsolete here, but why would we keep it public?

Comment (maxkatz6) [reply src/Avalonia.Controls/TopLevel.cs:306]: Its public members were removed, and it doesn't work with other input types.

File: src/Avalonia.Controls/Primitives/PopupPositioning/IPopupPositioner.cs [hunk 2]
```diff
@@ -452,16 +453,14 @@ static class PopupPositionerExtensions
...
-            var pointer = topLevel.PointToClient(topLevel.PlatformImpl?.MouseDevice.Position ?? default);
-#pragma warning restore CS0618 // Type or member is obsolete
-
             positionerParameters.Offset = offset;
             positionerParameters.ConstraintAdjustment = constraintAdjustment;
             if (placement == PlacementMode.Pointer)
             {
-                positionerParameters.AnchorRectangle = new Rect(pointer, new Size(1, 1));
+                // We need a better way for tracking the last pointer position
+                var position = topLevel.PointToClient(topLevel.LastPointerPosition ?? default);
```
Comment (maxkatz6) [inline src/Avalonia.Controls/Primitives/PopupPositioning/IPopupPositioner.cs:461]: I couldn't find a better idea, except keeping old property but internal.
I don't think we can really subscribe on popup's target' root pointer movements, as there might not be any movements between assigning the target and opening the popup.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022387-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by kekekeks — APPROVED** [review]

**Pull request merged**